scipy/_lib/tests/test_public_api.py

```diff
@@ -445,17 +445,10 @@ def test_private_but_present_deprecation(module_name, correct_module):
     else:
         import_name = f'scipy.{module_name.split(".")[1]}.{correct_module}'
 
-    correct_import = import_module(import_name)
-
     # Attributes that were formerly in `module_name` can still be imported from
-    # `module_name`, albeit with a deprecation warning. The specific message
-    # depends on whether the attribute is public in `scipy.xxx` or not.
+    # `module_name`, albeit with a deprecation warning.
     for attr_name in module.__all__:
-        attr = getattr(correct_import, attr_name, None)
-        if attr is None:
-            message = f"`{module_name}.{attr_name}` is deprecated..."
```

Comment:
Perhaps we should consider failing the test if we hit this branch so to ensure we haven't missed anything

Reply:
ie
```
attr = getattr(correct_import, attr_name, None)
assert attr is not None
message = f"Please import `{attr_name}` from the `{import_name}`..."
        with pytest.deprecated_call(match=message):
```

Reply:
Done. 👍